### DISCUSSION TITLE
Contributor call: 15 April 2023

### DISCUSSION THREAD

**Discussion by joostdecock** [freesewing/freesewing#3768] [Support] [answered]
You are all invited to the FreeSewing Contributor Call.
Please leave a comment here or [on Discord](https://discord.freesewing.org/) to add things to the agenda.
This issue will be updated after the call with notes and to-dos.

### About
FreeSewing's contributor call is an informal hangout/chat/meeting that is open to all FreeSewing contributors and users alike.
The call is organized/hosted by @tangerineshark.
The call does not have a fixed structure or agenda items. Everyone is free to suggest agenda items, ask questions, or bring ideas.

**Details**
Location: https://discord.freesewing.org (meeting room: 🔊`voice chat`)
Date: April 15 2023 ⚠**4pm**⚠ UTC, which can be found in your local time zone [here](https://savvytime.com/converter/utc-to-belgium-antwerpen-ca-los-angeles-il-chicago-ny-new-york-city-brazil-sao-paulo-united-kingdom-london-france-paris-south-africa-johannesburg-msd-india-mumbai-hkt-japan-tokyo-australia-sydney/apr-1-2023/4pm).
Or check the Events tab in Discord!
The meetings will always be at 6pm, Antwerp time.
⚠Watch out for timezone changes!

### Agenda
- FreeSewing has no middle management
- What's a bug and what's something that's not implemented yet? (see #3755 
-  #1102 / #3493
- I couldn't clearly find the list of issues needed to release v3. I'd love to help getting this new version out and a list of those issues would help quite a lot.
- I too, would like a (very brief) summary of where we are on the v3 roadmap, what the estimated timeframe might be for switchover, what needs to be done, etc.
-  Is there consensus that the correct and expected behavior of the FreeSewing UI is, when complete = false and paperless = true: 1. dimensions are hidden; and 2. the grid is still visible?
-    Is there interest in standardizing the behavior of plugins and their ability to display/hide themselves based on the complete setting? Is this something we want to pursue further?
- What is the UI/UX vision (for lack of a better term) for v3?
For the short interaction I had with lab, the UX doesn't feel as nice as the current v2.
- #3341 
- How do people feel about "non fashion" patterns, my background and interest is in the MYOG (Make your own gear) community which is generally focused on outdoor gear (backpack hammocks, bikepacking bags etc) and I am curious if that fits into the vibe of freesewing
- #3625 and #3819 
- I'd like to discuss not drafting top level parts that are hidden by a hide: {self: true} configuration, and not drafting their hidden dependencies if the whole chain is hidden.

**Comment by bobgeorgethe3rd** [comment]
Could we add  #1102 to the call please. I think I might have the solution for it but I don't if this is what #3493 is about or if that PR conflicts with the issue.

**Comment by luizfzs** [comment]
I couldn't clearly find the list of issues **needed** to release v3. I'd love to help getting this new version out and a list of those issues would help quite a lot. 

Could we add outlining the required work as a topic for discussion?

**Reply by BenJamesBen** [reply]
I, too, would like a (very brief) summary of where we are on the v3 roadmap, what the estimated timeframe might be for switchover, what needs to be done, etc.

**Comment by BenJamesBen** [comment]
Agenda items:
1. Is there consensus that the correct and expected behavior of the FreeSewing UI is, when `complete = false` and `paperless = true`: 1. dimensions are hidden; and 2. the grid is still visible? 
2. Is there interest in standardizing the behavior of plugins and their ability to display/hide themselves based on the `complete` setting? Is this something we want to pursue further?

**Comment by SeaZeeZee** [comment]
I would like to check in about #3341

**Comment by luizfzs** [comment]
What is the UI/UX vision (for lack of a better term) for v3?
For the short interaction I had with `lab`, the UX doesn't feel as nice as the current v2.

**Comment by joostdecock** [comment]
I've updated the agenda with the inputs so far. Keep 'em coming if there's more you (anyone) would like to discuss.

**Comment by clegganator259** [comment]
How do people feel about "non fashion" patterns, my background and interest is in the MYOG (Make your own gear) community which is generally focused on outdoor gear (backpack hammocks, bikepacking bags etc) and I am curious if that fits into the vibe of freesewing

**Reply by joostdecock** [reply]
Go for it. We are open to all sorts of patterns 🙌

**Comment by eriese** [comment]
I'd like to discuss not drafting top level parts that are hidden by a `hide: {self: true}` configuration, and not drafting their hidden dependencies if the whole chain is hidden. 

Firstly it's more efficient, but secondly it's helpful if your hidden top level part is inheriting from a part that expects certain things to be present in the stores that, due to settings, have not been stored.

To support this, I'd also like to discuss allow passing a function to the `hide` options that has, at the very least, the current `settings` as an argument. This will allow a designer to programmatically hide a part and/or its dependencies without drafting

**Comment by tangerineshark** [comment, marked as answer]
### Notes

Note on Notes: I caught up on these very heckin' late, apologies. They're pretty cursory, but feel free to ping me if you have follow-up questions and i'll point you in the right direction probably ~ @tangerineshark 

Welcome to some new folks! Clegganator and Luiz joined the call for the first time, welcome to both!

FreeSewing has no middle management - in general, make sure you're trying to say yes to people asking to contribute

What's a bug and what's something that's not implemented yet? (see https://github.com/freesewing/freesewing/pull/3755
https://github.com/freesewing/freesewing/issues/1102 / https://github.com/freesewing/freesewing/pull/3493

I couldn't clearly find the list of issues needed to release v3. I'd love to help getting this new version out and a list of those issues would help quite a lot. 

I too, would like a (very brief) summary of where we are on the v3 roadmap, what the estimated timeframe might be for switchover, what needs to be done, etc.

- for three above items, see discussion of v3 roadmap here: https://github.com/freesewing/freesewing/discussions/3934

Is there consensus that the correct and expected behavior of the FreeSewing UI is, when complete = false and paperless = true: 1. dimensions are hidden; and 2. the grid is still visible?

Is there interest in standardizing the behavior of plugins and their ability to display/hide themselves based on the complete setting? Is this something we want to pursue further?

- changing snippets in a way that will make it easier to override the plugin-bundle (e.g., with your own logo)

What is the UI/UX vision (for lack of a better term) for v3?

- check in soon: alpha of v3 is planned in a matter of weeks ideally

For the short interaction I had with lab, the UX doesn't feel as nice as the current v2.
https://github.com/freesewing/freesewing/issues/3341

How do people feel about "non fashion" patterns, my background and interest is in the MYOG (Make your own gear) community which is generally focused on outdoor gear (backpack hammocks, bikepacking bags etc) and I am curious if that fits into the vibe of freesewing
https://github.com/freesewing/freesewing/pull/3625 and https://github.com/freesewing/freesewing/issues/3819

- heck yeah go for it

I'd like to discuss not drafting top level parts that are hidden by a hide: {self: true} configuration, and not drafting their hidden dependencies if the whole chain is hidden.

**Comment by joostdecock** [comment]
Closing to increase visibility of active discussions
